geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java

```diff
@@ -365,6 +365,28 @@ public void testHSetNXExecutor() {
 
   }
 
+  @Test
+  public void hsetNX_shouldThrowErrorIfKeyIsWrongType() {
+    String string_key = "String_Key";
+    String set_key = "Set_Key";
+    String field = "field";
+    String value = "value";
+
+    jedis.set(string_key, value);
+    jedis.sadd(set_key, field);
+
+    assertThatThrownBy(
+        () -> jedis.hsetnx(string_key, field, "something else"))
+            .isInstanceOf(JedisDataException.class)
+            .hasMessageContaining("WRONGTYPE");
```

Comment:
It would be nice if we checked the full error message here.  We store the rest of it as a constant in `RedisConstants`, so it could be something like:
```
.hasMessageContaining("WRONGTYPE " + ERROR_WRONG_TYPE);
```

if you want to be even more exact, it could be:
```
.hasMessage("(error) WRONGTYPE " + ERROR_WRONG_TYPE);
```

I know it might seem like overkill, but I remember us having problems with other error messages not matching Redis's exactly.  There are other tests where we just check that the message contains `WRONGTYPE`, maybe we could change those as well.

Reply:
Good call on the argument count check, I did that for the HSTRLEN PR that's currently pending, too!

Think I might add a story for the WRONGTYPE messages, since that will touch a bunch of otherwise-unrelated commands.